src/mds/query.py

```diff
@@ -83,10 +83,9 @@ def add_filter(query):
                 query = query.where(Metadata.data[path].has_key(field))
             else:
                 values = ["*" if v == "\*" else v for v in values]
+                path = path if path == "_guid_type" else list(path.split("."))
```

Comment:
So is the performance been impacted because there is a list got passed as indices?
If so, instead of only looking for the _guid_type field, shouldn't we make sure if don't use a list if there is no . in the path for any field names?

Reply:
yeah, hold on, this worked in the specific case where we want to filter on guid_type, but this presumably breaks for records that don't have that field

Reply:
we populate that as a common pattern but it's not required, so I think we need to be a little more careful here

Reply:
could we just not pass a list if it's a single item without "."?

Reply:
`path = path if "." not in path else list(path.split("."))`

Reply:
then it will work for any query with a single item path

Reply:
The reason I was trying to propose adding an index to `_guid_type` is because we hardcode the MDS query from Data portal. --> [code here](https://github.com/uc-cdis/data-portal/blob/5f3dfe2e2f15a0b7527f8d6dcbc10b6fa4ec6d98/src/Discovery/Utils/MDSUtils/MDSUtils.jsx#L12C1-L12C100)

Reply:
yeah, hard-coding usage on the frontend is one thing, but hard-coding those patterns on the backend means every frontend and client (for which ours is not the only one) would need to behave with the same assumptions. We don't _want_ to enforce a particular pattern of usage unless that structure is built into the API itself and the expectations are clear. 

So I still think the general string is better. It means _if_ there is an index for the particular field being searched, it will be used (which means better performance beyond just the _guid_type if someone has indexed other columns). Technically MDS has an endpoint to index columns, though I'm not sure how well used that is